src/cljs/witan/ui/controllers/user.cljs

```diff
@@ -127,6 +161,23 @@
                            search) groups)]
     (data/swap-app-state! :app/user assoc :user/group-search-filtered filtered-groups)))
 
+(defmethod handle :signup
+  [event {:keys [username password] :as payload}]
+  (if-let [error (cond
+                   (not (apply = username)) :string/sign-up-error-usernames-match
+                   (not (apply = password)) :string/sign-up-error-passwords-match
```

Comment:
`username` and `password` are actually seqs because the confirmation check (do the passwords match? do the usernames match?) is done by controller.